integrations/sonarqube/client.py

```diff
@@ -256,17 +292,27 @@ async def get_issues_by_component(
         """
         component_issues = []
         component_key = component.get("key")
+        query_params = {"componentKeys": component_key}
         logger.info(f"Fetching all issues in component: {component_key}")
+
+        selector = typing.cast(
+            SonarQubeIssueResourceConfig, event.resource_config
+        ).selector
+
+        if hasattr(selector, "api_query_params") and selector.api_query_params:
+            selector_query_params = selector.api_query_params.generate_request_params()
+            query_params.update(selector_query_params)
+
         response = await self.send_paginated_api_request(
             endpoint=Endpoints.ISSUES,
             data_key="issues",
-            query_params={"componentKeys": component_key},
+            query_params=query_params,
         )
 
         for issue in response:
-            issue["__link"] = (
-                f"{self.base_url}/project/issues?open={issue.get('key')}&id={component_key}"
-            )
+            issue[
+                "__link"
```

Comment:
I know this isn't new code but, I think it's best to use a constant for these types of custom properties, WDYT?

Reply:
that's great and something we'll need to discuss with Yair and Tom as we decided to distinguish custom properties from the one returned by the third party API using __<propName>

Reply:
The property name is great!
I only thought since your using it in a lot of places it would be best to save it in a constant for reusability

Reply:
ahh got it now. Yeah, I'll make it constant